crates/bevy_ecs/src/component.rs

```diff
@@ -1298,6 +1623,18 @@ impl<T: Component> FromWorld for InitComponentId<T> {
     }
 }
 
+/// An error returned when the registration of a required component fails.
+#[derive(Error, Debug)]
+#[non_exhaustive]
+pub enum RequiredComponentsError {
+    /// The component is already a directly required component for the requiree.
+    #[error("Component {0:?} is already a directly required component for {1:?}")]
+    DuplicateRegistration(ComponentId, ComponentId),
+    /// An archetype with the component that requires other components already exists
+    #[error("An archetype with the component that requires other components already exists")]
+    ArchetypeExists(ComponentId),
```

Comment:
This error message should mention the component.

Reply:
Added